Packs/FortiGate/Integrations/FortiGate/FortiGate.py

```diff
@@ -103,1209 +2141,3696 @@ def create_addr_string(list_of_addr_data_dicts):
+
+    Returns:
+        CommandResults: Outputs of the command that represent an entry in warroom.
+    """
+    protocol_type = get_service_type(args)
+
+    name = args.get("serviceName", "")
+    vdom = args.get("vdom", DEFAULT_VDOM)
+    comment = args.get("comment")
+    category = args.get("category")
+    start_ip = args.get("start_ip")
+    end_ip = args.get("end_ip")
+    fqdn = args.get("fqdn")
+    tcp_port_ranges = argToList(args.get("tcpRange"))
+    udp_port_ranges = argToList(args.get("udpRange"))
+    sctp_port_ranges = argToList(args.get("sctpRange"))
+    icmp_type = arg_to_number(args.get("icmp_type"))
+    icmp_code = arg_to_number(args.get("icmp_code"))
+    ip_protocol = arg_to_number(args.get("ip_protocol"))
+
+    validate_optional_ipv4_addresses(start_ip, end_ip)
+
+    response = client.create_firewall_service(
+        name=name,
+        vdom=vdom,
+        comment=comment,
+        category=category,
+        protocol_type=protocol_type,
+        start_ip=start_ip,
+        end_ip=end_ip,
+        fqdn=fqdn,
+        tcp_port_ranges=tcp_port_ranges,
+        udp_port_ranges=udp_port_ranges,
+        sctp_port_ranges=sctp_port_ranges,
+        icmp_type=icmp_type,
+        icmp_code=icmp_code,
+        ip_protocol=ip_protocol,
+    )
+
+    outputs = build_service_outputs(args)
+    readable_output = f"## The firewall service '{name}' was successfully created."
+
+    return CommandResults(
+        outputs_prefix=SERVICE_CONTEXT,
+        outputs_key_field="Name",
+        outputs=outputs,
+        readable_output=readable_output,
+        raw_response=response,
+    )
+
+
+@logger
+def update_firewall_service_command(client: Client, args: dict[str, Any]) -> CommandResults:
+    """Update firewall services.
+
+    Args:
+        client (Client): Session to Fortigate to run API requests.
+        args (dict[str, Any]): Arguments passed down by the CLI to configure the request.
+
+    Raises:
+        DemistoException: If the input protocol type is different from the API.
+        DemistoException: If `action` was given without arguments TCP/UDP/SCTP parameters.
+
+    Returns:
+        CommandResults: Outputs of the command that represent an entry in warroom.
+    """
+    name = args.get("name", "")
+    vdom = args.get("vdom", DEFAULT_VDOM)
+    comment = args.get("comment")
+    category = args.get("category")
+    start_ip = args.get("start_ip")
+    end_ip = args.get("end_ip")
+    fqdn = args.get("fqdn")
+    tcp_port_ranges = argToList(args.get("tcpRange"))
+    udp_port_ranges = argToList(args.get("udpRange"))
+    sctp_port_ranges = argToList(args.get("sctpRange"))
+    icmp_type = arg_to_number(args.get("icmp_type"))
+    icmp_code = arg_to_number(args.get("icmp_code"))
+    ip_protocol = arg_to_number(args.get("ip_protocol"))
+    action = args.get("action")
+
+    input_protocol_type = None
+    response = client.list_firewall_services(name, vdom)
```

Comment:
The list call is necessary to validate the current protocol type and  maintain members in `TCP/UDP/SCTP`.
When updating a protocol in GUI it would work, but through the API an error is returned.